File: src/some_examples_cpp/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

#include <chrono>

using namespace std::chrono_literals;

class SimplePublisher : public rclcpp::Node
{   
public:
    SimplePublisher() : Node("simple_publisher"), count_ (0)
    {
        pub_ = create_publisher<std_msgs::msg::String>("topicpp", 10);
        timer_ = create_wall_timer(1s, std::bind(&SimplePublisher::timer_callback, this));

        RCLCPP_INFO(get_logger(), "Simple Publisher is publishing at 1Hz.");
    }

    void timer_callback()
    {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello, M! - counter  = " + std::to_string(count_++);
        RCLCPP_INFO(get_logger(), "Publishing: '%s'", msg.data.c_str());
        pub_->publish(msg);
    }
private:
    unsigned int count_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: some_examples_py/some_examples_py/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimplePublisher(Node):

    def __init__(self):
        super().__init__('simple_publisher')
        self.publisher_ = self.create_publisher(String, 'topicpy', 10)
        self.count = 0
        timer_period = 1.0  # seconds
        self.get_logger().info('Publishing after every %s seconds' % timer_period)
        self.timer = self.create_timer(timer_period, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = 'Hello M! - number %d' % self.count
        self.publisher_.publish(msg)
        self.get_logger().info('Publishing: "%s"' % msg.data)
        self.count += 1


def main():
    rclpy.init()
    simple_publisher = SimplePublisher()
    rclpy.spin(simple_publisher)
    simple_publisher.destroy_node()
    rclpy.shutdown()
    
if __name__ == '__main__':
    main()
    